### ISSUE TITLE
The toolbar does not show up on initial visit/load page

### ISSUE THREAD

**Issue by MilaMurry** [KristhianX/essential-buttons-toolbar#43]
Often times the toolbar will not show up on certain sites during first/initial visit but reloading the page does makes the toolbar shows up. Looking at the source code, Ive noticed some issues that could affect the behaviors.

1. `tryAppend` might call itself aka recursions which is very uneffective in this case.
2.  Each time it fails to append, it does not resolve itself which means other tasks in queue wont and do not have a chance to run.

I'd recommend doing it like this

```javascript
function sleep(duration) {
  return new Promise(resolve => setTimeout(resolve, duration));
}

// An async function will always return a Promise.
async function appendToolbar() {
  const maxRetries = 8
  const initialDelay = 100
  const backoffFactor = 2

  for (let retryCount = 0; retryCount <= maxRetries; retryCount++) {
    await new Promise((resolve, reject) => {
      if (document.body) {
        appendToolbarAndResolve(resolve)
        break
      } else if (retryCount < maxRetries) {
        resolve()
      } else {
        reject(new Error('Toolbar appending failed'))
        break
      }
    });
    // A delay for each iteration.
    const delay = initialDelay * Math.pow(backoffFactor, retryCount)
    await sleep(delay) // Holt the iteration until sleep is done & in the meantime let other tasks in queue to run.
  }
}
```

Later in  `initializeToolbar()` you can just call it llike this:
```javascript
async function initializeToolbar() {
  ...
  if (!isCurrentPageExcluded) {
    appendToolbar()
    ...
  }
}

**Comment by KristhianX** [comment]
Thanks for your suggestion.

I checked the `tryAppend` function, and it is working correctly. If it does not find the document body on the first try, it will retry at increasing intervals—100ms, 200ms, 400ms, and so on—until reaching the maximum number of retries. (This is done to allow slow pages to load and ensure the add-on does not interfere much by constantly checking.)

Previously, the maximum number of retries was lower—I don't remember the exact number—but in a previous update, I increased it to 8 retries because I encountered websites that were taking longer to load than the maximum time the add-on allowed for appending the toolbar. (At the time, I was using a slow internet connection.)

I will increase the number of retries to 10 for now, but I will also try to improve the code—perhaps by using a MutationObserver—to avoid having to check multiple times.
